src/url.rs

```diff
@@ -79,6 +79,17 @@ pub fn parse_data_url(url: &Url) -> (String, String, Vec<u8>) {
     (media_type, charset, blob)
 }
 
+pub fn referer_url(url: Url) -> Url {
```

Comment:
Maybe something like `get_referer_url`, `format_referer_url`, or `url_to_referer_url` would be more descriptive?  Just to be more consistent with function names in that file.